test/unit/rotate.js

```diff
@@ -50,6 +50,57 @@ describe('Rotation', function () {
             fixtures.assertSimilar(expectedOutput, data, done);
           });
       });
+
+      it(`${orientation} image with EXIF Orientation ${exifTag}: Auto-orient (alias of .rotate())`, function (done) {
+        const [expectedWidth, expectedHeight] = orientation === 'Landscape' ? [600, 450] : [450, 600];
+        sharp(input)
+          .autoOrient()
+          .toBuffer(function (err, data, info) {
+            if (err) throw err;
+            assert.strictEqual(info.width, expectedWidth);
+            assert.strictEqual(info.height, expectedHeight);
+            fixtures.assertSimilar(expectedOutput, data, done);
+          });
+      });
+
+      [90, 180, 270, 45].forEach(function (angle) {
+        const [inputWidth, inputHeight] = orientation === 'Landscape' ? [600, 450] : [450, 600];
+        const expectedOutput = fixtures.expected(`${orientation}_${exifTag}_rotate${angle}-out.jpg`);
+        it(`${orientation} image with EXIF Orientation ${exifTag}: Auto-rotate then rotate ${angle}`, function (done) {
+          const [width, height] = angle === 45 ? [742, 742] : [inputWidth, inputHeight];
+          const [expectedWidth, expectedHeight] = angle % 180 === 0 ? [width, height] : [height, width];
+
+          sharp(input)
+            .rotate()
+            .rotate(angle)
```

Comment:
Some level-setting. I want to make sure we are on the same page, @lovell.

If I read your comments in the proposal correctly, you expect that this method of auto-orienting and then adding an additional rotation should already  work.

If you run these tests, you'll see it only works for the 45 degree angle.

Whether you implied it should already work today or not, I think your preference is that this _actually is_ the correct way to achieve the goals of my proposal.